.travis.yml

```diff
@@ -13,25 +18,40 @@ env:
 
 jobs:
   allow_failures:
-    - rust: nightly
+    - stage: "Build and Test"
+      rust: nightly
   fast_finish: true
   include:
     - name: "Rust: stable - Examples"
+      stage: "Build and Test"
       language: rust
       rust: stable
       services:
         - docker
       install:
-        - (cd examples/truffle; yarn install)
         - docker pull trufflesuite/ganache-cli:v6.7.0
       script:
-        - (cd examples/truffle; yarn run build)
         - docker run -d -p 9545:8545 trufflesuite/ganache-cli:v6.7.0
         - cargo run --example async
         - cargo run --example linked
         - cargo run --package examples-generate
         - cargo run --example rinkeby
 
+    - name: "crates.io"
+      stage: "Deploy"
+      language: rust
+      rust: stable
+      env:
+        - RUSTC_WRAPPER=
```

Comment:
Why no caching here?

Reply:
More of a just in case - I am paranoid. Also I figured deployment happens rarely enough that cache speedup isn't worth it.

Reply:
Oh, a little background information - `cargo publish` does a full build before publishing. I am also unsure about `RUSTC_WRAPPER` interaction with `cargo publish`